src/native/managed/compile-native.proj

```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.Build.Traversal">
     <PropertyGroup>
         <!-- We always want to use release for publishing using NativeAOT -->
-        <NativeLibsPublishConfiguration>Release</NativeLibsPublishConfiguration>
+        <NativeLibsPublishConfiguration>$(Configuration)</NativeLibsPublishConfiguration>
```

Comment:
nit: remove the comment, if we're just going to pass `$(Configuration)` through.
nit2: we dont' need `NativeLibsPublishConfiguration` at all.  Just add `$(Configuration)` to the `SubprojectProps`